reactor-test/src/main/java/reactor/test/util/LoggerUtils.java

```diff
@@ -0,0 +1,281 @@
+
+/**
+ * This class eases testing interested in what reactor classes emit using {@link Logger loggers}.
+ *
+ * @author Eric Bottard
+ */
+public final class LoggerUtils {
+	private static Logger testLogger;
+
+	private LoggerUtils() {
+	}
+
+	/**
+	 * Sets a {@link Loggers#useCustomLoggers(Function) logger factory} that will return loggers that not only use the
+	 * original logging framework used by reactor, but also use the logger set via {@link #enableCaptureWith(Logger)}, irrespective
+	 * of its name or how it was obtained. The expectation here is that tests that want to assess that something is
+	 * logged by reactor will pass a {@link TestLogger} instance to {@link #enableCaptureWith(Logger)}, trigger the operation
+	 * under scrutiny, assert the logger contents and reset state by calling {@link #disableCapture()}.
+	 *
+	 * <p>This method should be called very early in the application/tests lifecycle, before reactor classes have created
+	 * their loggers.</p>
+	 *
+	 * @return a disposable that re-installs the original factory when disposed
+	 */
+	public static Disposable useCurrentLoggersWithCapture() {
+		try {
+			CapturingFactory capturingFactory = new CapturingFactory();
+			Loggers.useCustomLoggers(capturingFactory);
+			return capturingFactory;
+		}
+		catch (NoSuchFieldException | IllegalAccessException | NoSuchMethodException e) {
+			throw new RuntimeException("Could not install custom logger", e);
+		}
+	}
+
+	private static class CapturingFactory implements Function<String, Logger>, Disposable {
+
+		private final Method originalFactoryMethod;
+
+		private final Object orginalFactory;
+
+		private CapturingFactory() throws NoSuchFieldException, IllegalAccessException, NoSuchMethodException {
+			Field lfField = Loggers.class.getDeclaredField("LOGGER_FACTORY");
+			lfField.setAccessible(true);
+			orginalFactory = lfField.get(Loggers.class);
+			originalFactoryMethod = orginalFactory.getClass().getMethod("getLogger", String.class);
+			originalFactoryMethod.setAccessible(true);
+		}
+
+		@Override
+		public Logger apply(String category) {
+			try {
+				Logger original = (Logger) originalFactoryMethod.invoke(orginalFactory, category);
+				return new DivertingLogger(original);
+			}
+			catch (IllegalAccessException | InvocationTargetException e) {
+				throw new RuntimeException(e);
+			}
+		}
+
+		@Override
+		public void dispose() {
+			try {
+				Field lfField = Loggers.class.getDeclaredField("LOGGER_FACTORY");
+				lfField.setAccessible(true);
+				Object o = lfField.get(Loggers.class);
+
+				if (!(Loggers.getLogger(LoggerUtils.class) instanceof DivertingLogger)) {
+					throw new IllegalStateException("Expected the current factory to be " + this + ", found " + o + " instead");
+				}
+				lfField.set(Loggers.class, orginalFactory);
+			}
+			catch (NoSuchFieldException | IllegalAccessException e) {
+				throw new RuntimeException(e);
+			}
+		}
+	}
+
+	public static void enableCaptureWith(Logger testLogger) {
```

Comment:
nitpick: I'd put the static methods above both static inner classes, and add basic documentation